beacon-chain/rpc/eth/validator/validator.go

```diff
@@ -1022,6 +1025,13 @@ func (vs *Server) GetLiveness(ctx context.Context, req *ethpbv2.GetLivenessReque
 		Data: make([]*ethpbv2.GetLivenessResponse_Liveness, len(req.Index)),
 	}
 	for i, vi := range req.Index {
+		_, err = st.ValidatorAtIndex(vi)
```

Comment:
This is a bit of an expensive operation. It makes a copy of the validator and if you are iterating over a large request then you might have a noticeable burden on the server.

I think you could simply check that `vi` is less than `len(participation)` and return the out of range error if it fails that check